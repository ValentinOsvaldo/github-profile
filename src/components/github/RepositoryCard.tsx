import { useState } from 'react';
import { GitBranch, Star } from 'lucide-react';
import {
  Card,
  CardContent,
  CardDescription,
  CardFooter,
  CardHeader,
  CardTitle,
} from '@/components/ui/card';
import { Repository } from '@/interfaces/github/repository';
import { Badge } from '../ui/badge';

interface Props {
  repository: Repository;
}

export const RepositoryCard: React.FC<Props> = ({ repository }) => {
  const [position, setPosition] = useState({ x: 0, y: 0 });
  const [isHovered, setIsHovered] = useState(false);

  const handleMouseMove = (event: React.MouseEvent<HTMLDivElement>) => {
    const rect = event.currentTarget.getBoundingClientRect();
    setPosition({
      x: event.clientX - rect.left,
      y: event.clientY - rect.top,
    });
  };

  return (
    <a
      href={repository.html_url}
      target="_blank"
      rel="noopener noreferrer"
      className="block h-full w-full"
    >
      <Card
        className="relative flex flex-col justify-between cursor-pointer h-full w-full"
        onMouseMove={handleMouseMove}
        onMouseEnter={() => setIsHovered(true)}
        onMouseLeave={() => setIsHovered(false)}
      >
        {isHovered && (
          <div
            className="pointer-events-none absolute inset-0 transition duration-300"
            style={{
              background: `radial-gradient(300px at ${position.x}px ${position.y}px, rgba(70, 123, 255, 0.15), transparent 80%)`,
            }}
          />
        )}
        <CardHeader>
          <CardTitle>{repository.name}</CardTitle>
          <CardDescription></CardDescription>
        </CardHeader>
        <CardContent>
          <div className="mb-2 text-muted-foreground">
            {repository.description ? repository.description : 'No description'}
          </div>

          <div className="flex flex-row gap-2">
            <Badge>
              {repository.language ? repository.language : 'Unknown'}
            </Badge>
          </div>
        </CardContent>
        <CardFooter className="flex flex-row gap-2 text-muted-foreground">
          <div className="flex items-center flex-row gap-2">
            <GitBranch className="w-4 h-4" />
            {repository.forks_count}
          </div>
          <div className="flex items-center flex-row gap-2">
            <Star className="w-4 h-4" />
            {repository.stargazers_count}
          </div>
          <div>{new Date(repository.updated_at).toDateString()}</div>
        </CardFooter>
      </Card>
    </a>
  );
};
