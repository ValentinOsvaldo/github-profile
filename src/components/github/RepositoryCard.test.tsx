import { render } from '@testing-library/react';
import { RepositoryCard } from './RepositoryCard';

const repository = {
  id: 625364800,
  node_id: 'R_kgDOJUZPQA',
  name: '08-sockets',
  full_name: 'ValentinOsvaldo/08-sockets',
  private: false,
  owner: {
    login: 'ValentinOsvaldo',
    id: 69062379,
    node_id: 'MDQ6VXNlcjY5MDYyMzc5',
    avatar_url: 'https://avatars.githubusercontent.com/u/69062379?v=4',
    gravatar_id: '',
    url: 'https://api.github.com/users/ValentinOsvaldo',
    html_url: 'https://github.com/ValentinOsvaldo',
    followers_url: 'https://api.github.com/users/ValentinOsvaldo/followers',
    following_url:
      'https://api.github.com/users/ValentinOsvaldo/following{/other_user}',
    gists_url: 'https://api.github.com/users/ValentinOsvaldo/gists{/gist_id}',
    starred_url:
      'https://api.github.com/users/ValentinOsvaldo/starred{/owner}{/repo}',
    subscriptions_url:
      'https://api.github.com/users/ValentinOsvaldo/subscriptions',
    organizations_url: 'https://api.github.com/users/ValentinOsvaldo/orgs',
    repos_url: 'https://api.github.com/users/ValentinOsvaldo/repos',
    events_url: 'https://api.github.com/users/ValentinOsvaldo/events{/privacy}',
    received_events_url:
      'https://api.github.com/users/ValentinOsvaldo/received_events',
    type: 'User',
    user_view_type: 'public',
    site_admin: false,
  },
  html_url: 'https://github.com/ValentinOsvaldo/08-sockets',
  description: 'NodeJs Course by Fernando Herrera, ',
  fork: false,
  url: 'https://api.github.com/repos/ValentinOsvaldo/08-sockets',
  forks_url: 'https://api.github.com/repos/ValentinOsvaldo/08-sockets/forks',
  keys_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/keys{/key_id}',
  collaborators_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/collaborators{/collaborator}',
  teams_url: 'https://api.github.com/repos/ValentinOsvaldo/08-sockets/teams',
  hooks_url: 'https://api.github.com/repos/ValentinOsvaldo/08-sockets/hooks',
  issue_events_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/issues/events{/number}',
  events_url: 'https://api.github.com/repos/ValentinOsvaldo/08-sockets/events',
  assignees_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/assignees{/user}',
  branches_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/branches{/branch}',
  tags_url: 'https://api.github.com/repos/ValentinOsvaldo/08-sockets/tags',
  blobs_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/git/blobs{/sha}',
  git_tags_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/git/tags{/sha}',
  git_refs_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/git/refs{/sha}',
  trees_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/git/trees{/sha}',
  statuses_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/statuses/{sha}',
  languages_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/languages',
  stargazers_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/stargazers',
  contributors_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/contributors',
  subscribers_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/subscribers',
  subscription_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/subscription',
  commits_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/commits{/sha}',
  git_commits_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/git/commits{/sha}',
  comments_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/comments{/number}',
  issue_comment_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/issues/comments{/number}',
  contents_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/contents/{+path}',
  compare_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/compare/{base}...{head}',
  merges_url: 'https://api.github.com/repos/ValentinOsvaldo/08-sockets/merges',
  archive_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/{archive_format}{/ref}',
  downloads_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/downloads',
  issues_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/issues{/number}',
  pulls_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/pulls{/number}',
  milestones_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/milestones{/number}',
  notifications_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/notifications{?since,all,participating}',
  labels_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/labels{/name}',
  releases_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/releases{/id}',
  deployments_url:
    'https://api.github.com/repos/ValentinOsvaldo/08-sockets/deployments',
  created_at: '2023-04-08T22:11:02Z',
  updated_at: '2023-04-08T22:16:47Z',
  pushed_at: '2023-04-08T22:16:44Z',
  git_url: 'git://github.com/ValentinOsvaldo/08-sockets.git',
  ssh_url: 'git@github.com:ValentinOsvaldo/08-sockets.git',
  clone_url: 'https://github.com/ValentinOsvaldo/08-sockets.git',
  svn_url: 'https://github.com/ValentinOsvaldo/08-sockets',
  homepage: null,
  size: 10,
  stargazers_count: 0,
  watchers_count: 0,
  language: 'JavaScript',
  has_issues: true,
  has_projects: true,
  has_downloads: true,
  has_wiki: true,
  has_pages: false,
  has_discussions: false,
  forks_count: 0,
  mirror_url: null,
  archived: false,
  disabled: false,
  open_issues_count: 0,
  license: null,
  allow_forking: true,
  is_template: false,
  web_commit_signoff_required: false,
  topics: [],
  visibility: 'public',
  forks: 0,
  open_issues: 0,
  watchers: 0,
  default_branch: 'main',
};

describe('Repository Card', () => {
  it('Render component', () => {
    const card = render(<RepositoryCard repository={repository} />);

    expect(card).toMatchSnapshot();
  });
});
