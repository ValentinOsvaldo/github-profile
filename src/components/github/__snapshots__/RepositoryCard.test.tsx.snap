// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Repository Card > Render component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="block h-full w-full"
        href="https://github.com/ValentinOsvaldo/08-sockets"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm relative flex flex-col justify-between cursor-pointer h-full w-full"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="text-2xl font-semibold leading-none tracking-tight"
            >
              08-sockets
            </h3>
            <p
              class="text-sm text-muted-foreground"
            />
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="mb-2 text-muted-foreground"
            >
              NodeJs Course by Fernando Herrera, 
            </div>
            <div
              class="flex flex-row gap-2"
            >
              <div
                class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground hover:bg-primary/80"
              >
                JavaScript
              </div>
            </div>
          </div>
          <div
            class="items-center p-6 pt-0 flex flex-row gap-2 text-muted-foreground"
          >
            <div
              class="flex items-center flex-row gap-2"
            >
              <svg
                class="lucide lucide-git-branch w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="6"
                  x2="6"
                  y1="3"
                  y2="15"
                />
                <circle
                  cx="18"
                  cy="6"
                  r="3"
                />
                <circle
                  cx="6"
                  cy="18"
                  r="3"
                />
                <path
                  d="M18 9a9 9 0 0 1-9 9"
                />
              </svg>
              0
            </div>
            <div
              class="flex items-center flex-row gap-2"
            >
              <svg
                class="lucide lucide-star w-4 h-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                />
              </svg>
              0
            </div>
            <div>
              Sat Apr 08 2023
            </div>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="block h-full w-full"
      href="https://github.com/ValentinOsvaldo/08-sockets"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm relative flex flex-col justify-between cursor-pointer h-full w-full"
      >
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
          >
            08-sockets
          </h3>
          <p
            class="text-sm text-muted-foreground"
          />
        </div>
        <div
          class="p-6 pt-0"
        >
          <div
            class="mb-2 text-muted-foreground"
          >
            NodeJs Course by Fernando Herrera, 
          </div>
          <div
            class="flex flex-row gap-2"
          >
            <div
              class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground hover:bg-primary/80"
            >
              JavaScript
            </div>
          </div>
        </div>
        <div
          class="items-center p-6 pt-0 flex flex-row gap-2 text-muted-foreground"
        >
          <div
            class="flex items-center flex-row gap-2"
          >
            <svg
              class="lucide lucide-git-branch w-4 h-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="6"
                x2="6"
                y1="3"
                y2="15"
              />
              <circle
                cx="18"
                cy="6"
                r="3"
              />
              <circle
                cx="6"
                cy="18"
                r="3"
              />
              <path
                d="M18 9a9 9 0 0 1-9 9"
              />
            </svg>
            0
          </div>
          <div
            class="flex items-center flex-row gap-2"
          >
            <svg
              class="lucide lucide-star w-4 h-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
              />
            </svg>
            0
          </div>
          <div>
            Sat Apr 08 2023
          </div>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
